src/Intl/Idn/Idn.php

```diff
@@ -516,6 +524,8 @@ private static function validateLabel($label, Info $info, array $options, $canBe
             if ('-' === substr($label, -1, 1)) {
                 $info->errors |= self::ERROR_TRAILING_HYPHEN;
             }
+        } elseif (0 === strncmp($label, 'xn--', 4)) {
```

Comment:
`'xn--' === substr($label, 0, 4)` to match the same check above (or replace the code above maybe?)